src/algebra/big_operators/basic.lean

```diff
@@ -1322,6 +1334,36 @@ lemma prod_finset_coe [comm_monoid β] :
   ∏ (i : (s : set α)), f i = ∏ i in s, f i :=
 (finset.prod_subtype s (λ _, iff.rfl) f).symm
 
+@[to_additive]
+lemma prod_unique {α β : Type*} [comm_monoid β] [unique α] (f : α → β) :
+  (∏ x : α, f x) = f (default α) :=
+by rw [univ_unique, prod_singleton]
+
+@[to_additive]
+lemma prod_subsingleton_univ {α β : Type*} [comm_monoid β] [subsingleton α] (f : α → β) (a : α) :
+  (∏ x : α, f x) = f a :=
+begin
+  haveI : unique α := { default := a, uniq := λ _, subsingleton.elim _ _ },
+  convert prod_unique f
+end
+
+@[simp] lemma prod_unique_has_one {α β : Type*} [unique α] [has_one α] [comm_monoid β]
+  (f : α → β) :
+  ∏ x : α, f x = f 1 :=
+by rw [prod_unique f, subsingleton.elim (default α) 1]
+
+@[simp] lemma sum_unique_has_one {α β : Type*} [unique α] [has_one α] [add_comm_monoid β]
```

Comment:
These seem like weird lemmas to me - why not just have a `default x = 0` lemma for `has_zero`?